src/OrchardCore.Modules/OrchardCore.ContentFields/Views/TextField-CodeMirror.Edit.cshtml

```diff
@@ -0,0 +1,33 @@
+@model OrchardCore.ContentFields.ViewModels.EditTextFieldViewModel
+
+@{
+    var settings = Model.PartFieldDefinition.GetSettings<TextFieldSettings>();
+    var culture = await Orchard.GetContentCultureAsync(Model.Field.ContentItem);
+}
+
+<div class="form-group">
+    <label asp-for="Text">@Model.PartFieldDefinition.DisplayName()</label>
+    <input asp-for="Text" class="form-control content-preview-text" dir="@culture.GetLanguageDirection()" />
+    @if (!String.IsNullOrEmpty(settings.Hint))
+    {
+        <span class="hint">@settings.Hint</span>
+    }
+</div>
+
+<script asp-name="codemirror" depends-on="admin" at="Foot"></script>
+<script asp-name="codemirror-mode-css" at="Foot"></script>
+<script asp-name="codemirror-mode-htmlmixed" at="Foot"></script>
+<script asp-name="codemirror-mode-javascript" at="Foot"></script>
+<script asp-name="codemirror-mode-xml" at="Foot"></script>
+<script at="Foot">
+    $(function () {
+        var optionsTexArea = document.getElementById('@Html.IdFor(x => x.Text)');
+        var editor = CodeMirror.fromTextArea(optionsTexArea, {
+            lineNumbers: false,
+            lineWrapping: true,
+            matchBrackets: true,
+            mode: { name: "htmlmixed" }
+        });
+        setTimeout(function() { editor.refresh(); },1);
+    });
```

Comment:
@agriffard I am still confused why we need the `setTimeout` here, but not other places?
is it because the editor option is initially hidden, and codemirror doesn't draw initially when unhidden?

Reply:
If the editor is not refreshed, it is not resized to show the whole text inside.
Uncomment this line and load the page with a long text to see what it does.

Reply:
`autoRefresh: true,` is the code mirror option we use elsewhere, and works for `Autoroute` (I will see later when I have time to see if it works here)